sw/device/silicon_owner/tock/tests/uart/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_main]
+#![no_std]
+
+use core::cell::Cell;
+use core::cmp::min;
+use libtock::console::Console;
+use libtock::platform::{share, AllowRo, DefaultConfig, Subscribe, Syscalls};
+use libtock::runtime::{set_main, stack_size, TockSyscalls};
+
+set_main!(main);
+stack_size!(0x800);
+
+// Reads data from the console input. The data is read in chunks whose length is
+// given by `lens`. The newly-read data is returned. Panics upon encountering an
+// error condition.
+fn read_exact<'b>(buffer: &'b mut [u8], lens: &[usize]) -> &'b [u8] {
+    assert!(buffer.len() >= lens.iter().sum());
+    let upcall: Cell<Option<(u32, u32)>> = Cell::new(None);
+    let mut cursor = 0;
+    share::scope(|handle| {
+        TockSyscalls::subscribe::<_, _, DefaultConfig, 1, 2>(handle, &upcall).unwrap();
+        for &len in lens {
+            let bytes = share::scope(|handle| {
+                TockSyscalls::allow_rw::<DefaultConfig, 1, 1>(handle, &mut buffer[cursor..])
+                    .unwrap();
+                if let Some(error) = TockSyscalls::command(1, 2, len as u32, 0).get_failure() {
```

Comment:
Are there symbolic constants for `driver_id` and `command_id` values?

Reply:
I added symbolic constants.